ambuda/static/js/main.js

```diff
@@ -331,14 +367,11 @@ const TextContent = (() => {
         const $undoParse = e.target.closest('.js--source');
         if ($undoParse) {
           e.preventDefault();
-          Server.getText(
-            $undoParse.href,
-            (resp) => {
-              const transResp = transliterateSanskritBlob(resp, Preferences.contentScript);
-              $undoParse.closest('s-lg').outerHTML = transResp;
-            },
-            () => {},
-          );
+          // Hide the right (parsed) block.
+          const parsedNode = $undoParse.closest('.block-parsed-right');
+          parsedNode.style.display = 'none';
+          const $block = parsedNode.previousSibling;
```

Comment:
That does seem like a good structure, but it might require significant refactoring of existing code to get it to work.

This is because, in order to implement it, I will need to change `to_verse` to have the following behavior:
- if getting the original version, it will need to wrap `s-lg` in `<div class="block">`
- if getting the parsed version, it should not wrap `s-lg` in `<div class="block">`

Feels like it may be too big a change for this PR, so perhaps we can revisit / rewrite how verse markup is generated in a future PR? What do you think?

Reply:
Or worth doing it here?

One possible solution is to leave `to_verse` alone and in the method for `/<text_slug>/<section_slug>` in texts.py, add some code that traverses the XML and replaces `s-lg` elements with `<div class="block">`. Does this seem like an appropriate solution for this PR, or do you think it's perhaps too unwieldy?

Reply:
Let's split it out -- I'll send you a PR with the markup update, then you can rebase on top with this one.

Reply:
See #131.